tests/kernel_bundle/sycl_compile.h

```diff
@@ -0,0 +1,163 @@
+    log.skip("No kernel bundle with input state with kernel: " + kName +
+             " (skipped).");
+    return;
+  }
+
+  auto input_kb =
+      sycl::get_kernel_bundle<sycl::bundle_state::input>(ctx, {dev}, {k_id});
+  auto input_ids = input_kb.get_kernel_ids();
+
+  auto obj_kb = compile_bundle<Overload>(input_kb, {dev});
+  kernel_bundle::define_kernel<DescriptorT, sycl::bundle_state::input>(queue);
+
+  auto restrictions{
+      kernel_bundle::get_restrictions<DescriptorT,
+                                      sycl::bundle_state::input>()};
+  bool dev_is_compat = restrictions.is_compatible(dev);
+
+  if (obj_kb.has_kernel(k_id) != dev_is_compat) {
+    FAIL(log, "Device does not support kernel " + kName +
+                  " but compiled bundle"
+                  " contains it");
+  }
+
+  // Check that result object bundle has the same kernels as input bundle
+  if (!std::any_of(input_ids.cbegin(), input_ids.cend(),
+                   [&](auto &in_id) { return obj_kb.has_kernel(in_id); })) {
+    FAIL(log,
+         "Result bundle does not contain all kernels from input bundle "
+         "(kernel: " +
+             kName + ")");
+  }
+
+  // Check that input and result kernel bundles have the same context
+  if (input_kb.get_context() != obj_kb.get_context()) {
+    FAIL(log,
+         "Input bundle and result bundle have different contexts "
+         "(kernel: " +
+             kName + ")");
+  }
+}
+
+/** @brief Check that sycl::compile result bundle has the same associated
+ *         devices as presented in input device vector
+ *  @tparam Overload Selected overload of sycl::compile
+ */
+template <CompileOverload Overload>
+static void check_associated_devices(util::logger &log) {
+  auto queue = util::get_cts_object::queue();
+  const auto ctx = queue.get_context();
+  std::vector<sycl::device> devices{queue.get_device()};
+  // Force duplicates
+  devices.push_back(devices[0]);
+
+  // Test can be skipped if all devices do not support online compilation
+  // since this is not sycl::compile's fault
+  if (!sycl::has_kernel_bundle<sycl::bundle_state::input>(ctx)) {
+    log.skip("No kernel bundle with input state for test (skipped).");
+    return;
+  }
+
+  const auto input_kb =
+      sycl::get_kernel_bundle<sycl::bundle_state::input>(ctx, devices);
+  const auto obj_kb = compile_bundle<Overload>(input_kb, devices);
+  const auto kb_devs = obj_kb.get_devices();
+
+  // Check that result kernel bundle contains all devices from passed vector
+  if (!std::equal(kb_devs.begin(), kb_devs.end(), devices.begin())) {
+    FAIL(log,
+         "Set of associated to obj_kb devices is not equal to list of "
+         "devices passed.");
+  }
+
+  // TODO: we can't use std::set<sycl::device> due to the fact that sycl::device
+  // has no comparative operators.
+  // When this issue will be resolved the following statement can be used insted
+  // of cycle with std::count calling:
```

Comment:
```suggestion
  // has no comparison operators.
  // Once this issue is resolved, the following statement can be used instead
  // of cycling with a std::count call:
```

Reply:
I've applied this comment in this commit: d910c659140ff7236d0fdc6281a733d02b2b9a3c